app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
```

Comment:
* Ugh. `GithubService.add_labels_to_an_issue` *does* create a new label if it doesn't exist. As you mentioned, for the GHNM it's fine because we check first (because we already have the issue!) but here would require another call. I'm tempted to say whatever, because as you said this isn't user configurable at all. We should still add our own method to GithubService as I'm sure we'll need the 'add label only if exists' functionality eventually anyway (not this PR).

* Adding a label to an issue that already had that label *is* idempotent, thankfully. No notification, no UI update of any variety.